tests/phpunit/Unit/functions/StrToSnakeCaseTest.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace Cig\Tests\Unit\Functions;
+
+
+class StrToSnakeCaseTest extends \Cig\Tests\Unit\BaseTestCase {
+
+	/**
+	 * @covers ::\Cig\str_to_snake_case()
+	 */
+	public function test_str_to_snake_case(): void {
+		$string = "welcome~ to *snake case!!";
+		$expected_result = "welcome_to_snake_case";
+
+		//this method uses another method (str_to_words) from same file
```

Comment:
Also like the other places `str_to_words` is called it should be stubbed along with anything else that isn't core PHP called inside `str_to_pascal_case`